pandas/_testing/_io.py

```diff
@@ -398,6 +400,14 @@ def write_to_compressed(compression, path, data, dest="test"):
         mode = "w"
         args = (dest, data)
         method = "writestr"
+    elif compression == "tar":
```

Comment:
I think this entire function could be replaced with a call to `get_handle` (not needed in this PR)